test/Jamfile.v2

```diff
@@ -30,6 +30,24 @@ rule change-test_update-exe-name ( name : type ? : property-set )
 project
     : requirements
       <library>/boost/multi_index//boost_multi_index
+      <library>/boost/assert//boost_assert
+      <library>/boost/bind//boost_bind
+      <library>/boost/config//boost_config
+      <library>/boost/container_hash//boost_container_hash
+      <library>/boost/core//boost_core
+      <library>/boost/foreach//boost_foreach
+      <library>/boost/integer//boost_integer
+      <library>/boost/iterator//boost_iterator
+      <library>/boost/move//boost_move
+      <library>/boost/mp11//boost_mp11
+      <library>/boost/mpl//boost_mpl
+      <library>/boost/preprocessor//boost_preprocessor
+      <library>/boost/smart_ptr//boost_smart_ptr
+      <library>/boost/static_assert//boost_static_assert
+      <library>/boost/throw_exception//boost_throw_exception
+      <library>/boost/tuple//boost_tuple
+      <library>/boost/type_traits//boost_type_traits
+      <library>/boost/utility//boost_utility
```

Comment:
Do you really use all these? Directly?

Reply:
You only need the direct dependencies here. --test gives you everything, including secondaries.

For the primary dependencies used by your tests, use `boostdep --track-tests --primary multi_index`, but that's going to also give you the dependencies of the multi_index headers, which you don't need to replicate here.

There's no boostdep command that will tell you what your tests need in addition to what you already get from `<library>multi_index`.

(From looking at the diff of the two boostdep outputs, I think that the answer is `foreach` and `serialization`.)

Reply:
Trimmed it down to
```
foreach
iterator
mpl
serialization
```
I wonder if `boostdep` should have an option to explicit calculate this, as it is exactly what Boost authors will need for their CML.txts/Jamfiles.